crates/runtime/src/dataaccelerator/duckdb.rs

```diff
@@ -104,9 +105,49 @@ impl DuckDBAccelerator {
 
         Some(
             self.duckdb_factory
-                .duckdb_file_path(&dataset.name.to_string(), &mut params),
+                .duckdb_file_path("accelerated_duckdb", &mut params),
         )
     }
+
+    pub async fn get_shared_pool(&self, dataset: &Dataset) -> Result<DuckDbConnectionPool> {
+        let duckdb_file = self.duckdb_file_path(dataset);
+
+        let acceleration = dataset
+            .acceleration
+            .as_ref()
+            .context(AccelerationNotEnabledSnafu {
+                dataset: dataset.name.to_string(),
+            })?;
+
+        let pool = match (acceleration.mode, duckdb_file) {
+            (Mode::File, Some(duckdb_file)) => self
+                .duckdb_factory
+                .get_or_init_file_instance(duckdb_file)
+                .await
+                .boxed()
+                .context(AccelerationCreationFailedSnafu)?,
+            (Mode::Memory, None) => self
+                .duckdb_factory
+                .get_or_init_memory_instance()
```

Comment:
nice, does this let us do in-memory federation as well?

Reply:
Yep!